pyresample/geometry.py

```diff
@@ -1105,10 +1105,16 @@ def _compute_bound_centers_dask(self, proj_dict, lons, lats):
             # cross anti-meridian of projection
             xarr_pos = da.where(xarr >= 0, xarr, np.nan)
             xarr_neg = da.where(xarr < 0, xarr, np.nan)
-            xmin = np.nanmin(xarr_pos)
-            xmax = np.nanmax(xarr_neg) + 360
-            xmin, xmax = da.compute(xmin, xmax)
-        return xmin, ymin, xmax, ymax
+            new_xmin = np.nanmin(xarr_pos)
+            new_xmax = np.nanmax(xarr_neg) + 360
+            if antimeridian_mode == "global_extents":
+                new_xmin, new_xmax = (-180.0, 180.0)
```

Comment:
@pnuu @mraspaud I'm looking for help on this case. In this method I'm supposed to return the bounds of the centers of pixels, but -180/180 are the outer extents of the pixels. The only way to know what the centers would be would be to handle the resolution but all the resolution handling is happened in another method after the call to this method.

I could use `None` or something to indicate to the later method "use the maximum extents possible", but that seems pretty error prone. I'm also realizing that I assume in this method that the input geographic (`+proj=latlong`) projection doesn't already have a `+pm=180` which would make the hardcoded -180/180 incorrect.

So my question is, how can I freeze the dynamic area definition to the global extents of the projection without making the code ugly by passing `resolution` down to this method?

Reply:
@djhoese TBH I do not fully understand what you are resolving here. But it seems to me that this might be related to the PyCoast horizontal scratch problem of cylindrical projections when crossing the dateline that I just solved with PR59. Does PyCoast after the changes you propose here see different dynamic area definition boundaries than what it gets to see now?

Reply:
Pycoast should never see a DynamicAreaDefinition object directly. This code doesn't change if you receive a valid area or not, but rather which of the potential valid areas you get. Do you have a specific Satpy use case that wasn't working?

Reply:
I had no specific case because TBH I have never tested a dynamic area def that crosses the datelline before. So I was just wondering whether my recent fix of the scratch problem in PyCoast would a) work at all and b) be possibly impaired by what you do here. I add two images of Metop-B that show at least a) with and without my fix to work as expected now. 
![Metop-Bsunny-20220510-DAY-2035-natural_with_night_fog-omerc_bb_1](https://user-images.githubusercontent.com/101883219/168020641-404ed305-bfb2-4be1-9a32-5c8ee04ea93f.jpg)
![Metop-Bsunny-20220510-DAY-2035-natural_with_night_fog-omerc_bb_2](https://user-images.githubusercontent.com/101883219/168020665-b264a05f-ae92-49bf-9035-9d69fce59ee4.jpg)

Reply:
I can't think of a reason why this DynamicAreaDefinition stuff would cause issues in pycoast. The result of using a DynamicAreaDefinition is a regular AreaDefinition. So the result of my work here won't specifically be the problem (the results are all valid areas), but it could reveal that something in pycoast still doesn't work in cases where AreaDefinitions:

1. Have a right-most extent >180 (this is what is currently done).
2. Have a projection with a prime meridian at 180. This is still valid PROJ, so pycoast shouldn't care too much unless it is doing `area_def.crs.is_geographic` and assuming that the prime meridian is 0.

I think those are the only special cases, but those are possible to create from a user perspective with or without this PR.

Reply:
In my above images PyCoast returns lon_min = 163.5 and lon_max = -154.5. Not sure what happens if lon_max > 180.

Reply:
Sometimes it depends exactly how pycoast is determining those values. Pycoast shouldn't assume much about the area and should be converting X/Y coordinates to normal lon/lat values. In these of these new areas these will also be lon/lat with a prime meridian at 180 instead of metered X/Y projection coordinates so...it should be OK. :crossed_fingers:

Reply:
To determine whether a (coasts, borders, rivers, ..) Polygon or Line for some projection and area_extent should be drawn or is 'irrelevant' (and potentially harmful !) PyCoast uses the GSHHG bounding boxes. These are strictly within [-180, 180].

Reply:
To the original question: aren't all extents the extreme edges of the pixels/data, not centers?

Reply:
@pnuu I've lost track of this conversation, but yes extents are the edges of the pixels, centers are the centers of the pixels. The method being changed here assumes that the lon/lat coordinates contained within the SwathDefinition are the centers of the pixels. I've updated the method to return `None` for the X corner coordinates if they should be considered to be the furthest/full extent of the projection.